eitprocessing/features/breath_detection.py

```diff
@@ -48,9 +53,17 @@ class BreathDetection:
     invalid_data_removal_percentile: int = 5
     invalid_data_removal_multiplier: int = 4
 
-    def find_breaths(self, sequence: Sequence, continuousdata_label: str) -> IntervalData:
+    @singledispatchmethod
+    def find_breaths(self, container: Sequence | ContinuousData) -> IntervalData:
         """Find breaths in the data.
 
+        You can either pass a ContinuousData object as first argument, or a
+        Sequence as first argument and the label of a ContinuousData object.
+        When passing a ContinuousData object, optionally, you can pass a
+        Sequence object as second argument.
+        If a sequence is passed to `find_breaths()`, the resulting breaths
+        are stored in the Sequence as `intervaldata["breaths"]`.
```

Comment:
I find this quite confusing. Both the explanation as well as the coding using the single dispatch method.

This explanation does not follow the way the code is actually defined, where a single argument of either `ContinuousData` or `Sequence` is expected. In practice, however, it seems that the expectation is one of 
1. ContinuousData (only)
2. ContinuousData + Sequence
3. Sequence + label.

The docstring is not explicit about what happens in each case (e.g. difference between 1 and 2 above).

I would prefer writing this function as follows (the argument names can be better, but just to give you the idea)

```python
    def find_breaths(
        self,
        detect_from: Sequence | ContinuousData,
        store_in: Sequence | None = None,
        label: str | None = None, # or some other default
    ) -> IntervalData:
```

I also would find it more readable if rather than overloading/single dispatch, there is a single function that starts with some conditionals to "mold" the inputs into an expected format from where it can proceed linearly.
This may just be a matter of opinion, though.

Reply:
If you don't agree with the `@singledispatch` method, I'd rather just not support this for now.
`detect_from` is way more confusing to me. 
Maybe the better way is to just accept a `ContinuousData` object, and accept an optional `Sequence` object for storing results. 

I like the `store_in` name. That would result in:

```python
bd = BreathDetection(...)
bd.find_breaths(sequence['global_impedance_(raw)'], store_in=sequence, store_label='breaths')
```

I like this approach better than `find_breaths(sequence, continuousdata_label=...)` to be honest.

@JulietteFrancovich, what do you think?

Reply:
I only now realize that there would be a separate `store_label` and `continuousdata_label` in that case, which indeed complicates the matter.
That being said, I do in principle like that you can either pass a `Sequence` or a `ContinuousData` as input, but it might be too much hassle to implement smoothly.

I don't think there's anything wrong with the singledispatch methodology. As I said, it's more of my personal preference for handling different cases with conditions. My main issue was that the types of inputs were unclear/inconsistent. 
If you do want to have "complex" combinbations of inputs, you can always make them keyword only to enforce clarity when using the function.

Reply:
I have removed the single dispatch. The function now only accepts a ContinuousData object as input. You can still provide a Sequence to store the data. I have re-requested a review.